src/main.rs

```diff
@@ -289,21 +296,20 @@ impl Backend {
             self.client.log_message(msg.message_type, msg.message).await;
         }
 
-        self.client
-            .publish_diagnostics(
-                uri.clone(),
-                messages
-                    .into_iter()
-                    .map(|(message, range)| Diagnostic {
+        for (uri, message, range) in messages.into_iter() {
+            self.client
+                .publish_diagnostics(
+                    uri,
+                    vec![Diagnostic {
                         range,
                         severity: Some(DiagnosticSeverity::ERROR),
                         message,
                         ..Default::default()
-                    })
-                    .collect(),
-                None,
-            )
-            .await;
+                    }],
+                    None,
+                )
+                .await;
+        }
```

Comment:
Can we use something like [`join_all`](https://docs.rs/futures/latest/futures/future/fn.join_all.html) so we don't have to wait for each promise individually?